src/auth_endpoints/signup.auth.ts

```diff
@@ -1,5 +1,44 @@
 import { Request, Response } from "express";
+import prismaClient from "../client.js";
+import crypto, { pbkdf2, pbkdf2Sync } from "crypto";
+import { validateRequestBody } from "../utils/auth/validate.request.body.js";
+import { hashPassword } from "../utils/auth/hash.password.js";
+import { isUsernameTaken } from "../utils/auth/username.availability.js";
```

Comment:
I can only allow using .ts import extensions if I am using a bundler, but I am use the default typescript compiler **TSC** so it has to be .js import because it will be ran as .js